lib/Dependency.js

```diff
@@ -52,6 +52,16 @@
  * @property {Module[]=} dependencies module on which the result depends on
  */
 
+/**
+ * @enum {number}
+ */
+const DependencyCategories = {
```

Comment:
I would prefer it to be just a string instead of an enum. enum would limit the possible values, but I would expect that plugins can add additional categories.